docs/en/writing-running-appium/tutorial/swipe/simple-element.md

```diff
@@ -0,0 +1,206 @@
+## Simple Element swipe
+
+### Android: UIAutomator2
+
+With UIAutomator2 driver, by default, we see only elements on screen. If
+element continues outside screen, then this part will be hidden.
+
+![simple-element-swipe.png](images/simple-element-swipe.png)
+
+```java
+/**
+ * Performs swipe inside element
+ *
+ * @param el  the swipe element
+ * @param dir the direction of swipe
+ * @version java-client: 7.3.0
+ **/
+public void swipeElementAndroid(MobileElement el, Direction dir) {
+    System.out.println("swipeElementAndroid(): dir: '" + dir + "'"); // always log your actions
+
+    // Animation default time:
+    //  - Android: 300 ms
+    //  - iOS: 200 ms
+    // final value depends on your app and could be larger
+    final int ANIMATION_TIME = 200; // ms
+
+    final int PRESS_TIME = 200; // ms
+
+    int edgeBorder;
+    PointOption pointOptionStart, pointOptionEnd;
+
+    // init screen variables
+    Rectangle rect = el.getRect();
+    // sometimes it is needed to configure edgeBorders
+    // you can also improve borders to have vertical/horizontal
+    // or left/right/up/down border variables
+    edgeBorder = 0;
+
+    switch (dir) {
+        case DOWN: // from up to down
+            pointOptionStart = PointOption.point(rect.x + rect.width / 2,
+                    rect.y + edgeBorder);
+            pointOptionEnd = PointOption.point(rect.x + +rect.width / 2,
+                    rect.y + rect.height - edgeBorder);
+            break;
+        case UP: // from down to up
+            pointOptionStart = PointOption.point(rect.x + rect.width / 2,
+                    rect.y + rect.height - edgeBorder);
+            pointOptionEnd = PointOption.point(rect.x + rect.width / 2,
+                    rect.y + edgeBorder);
+            break;
+        case LEFT: // from right to left
+            pointOptionStart = PointOption.point(rect.x + rect.width - edgeBorder,
+                    rect.y + rect.height / 2);
+            pointOptionEnd = PointOption.point(rect.x + edgeBorder,
+                    rect.y + rect.height / 2);
+            break;
+        case RIGHT: // from left to right
+            pointOptionStart = PointOption.point(rect.x + edgeBorder,
+                    rect.y + rect.height / 2);
+            pointOptionEnd = PointOption.point(rect.x + rect.width - edgeBorder,
+                    rect.y + rect.height / 2);
+            break;
+        default:
+            throw new IllegalArgumentException("swipeElementAndroid(): dir: '" + dir + "' NOT supported");
+    }
+
+    // execute swipe using TouchAction
+    try {
+        new TouchAction(driver)
+                .press(pointOptionStart)
+                // a bit more reliable when we add small wait
```

Comment:
Actually, Android is trying to be "smart" there and internally adds a delay on its own.

Normally, the pause between press and move has to be not lesser than 500ms (which equals to long tap pause. See https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/view/ViewConfiguration.java for more details)